crates/flux-fhir-analysis/src/wf/sortck.rs

```diff
@@ -336,30 +361,40 @@ impl<'genv, 'tcx> InferCtxt<'genv, 'tcx> {
                 self.check_expr(e2, &sort)?;
                 Ok(rty::Sort::Bool)
             }
-            fhir::BinOp::Mod => {
-                self.check_expr(e1, &rty::Sort::Int)?;
-                self.check_expr(e2, &rty::Sort::Int)?;
-                Ok(rty::Sort::Int)
-            }
+
             fhir::BinOp::Lt | fhir::BinOp::Le | fhir::BinOp::Gt | fhir::BinOp::Ge => {
                 let sort = self.next_sort_var();
                 self.check_expr(e1, &sort)?;
                 self.check_expr(e2, &sort)?;
 
-                // Elaborate sort of operator
-                let sort = self
-                    .fully_resolve(&sort)
-                    .map_err(|_| self.emit_err(errors::CannotInferSort::new(expr.span)))?;
+                // Elaborate sort of operator (force as `Int` for `tests/pos/surface/forall01.rs`)
+                let sort = self.resolve_or_unify(&sort, rty::Sort::Int);
```

Comment:
This feels hacky. We should do the same as we do for literals, put the sort in a temporary table, and resolve at the end. I think this should be enough for `forall01.rs` because the sort will be unified with the sort of `x`